src/main/java/org/threadly/litesockets/UDPClient.java

```diff
@@ -180,6 +176,31 @@ public boolean setSocketOption(final SocketOption so, final int value) {
   public ClientOptions clientOptions() {
     return uso;
   }
+  
+  @Override
+  protected void addReadBuffer(final ByteBuffer bb) {
+    addReadStats(bb.remaining());
+    synchronized(readerLock) {
+      readBuffers.add(bb);
+      callReader();
```

Comment:
thats a good point, didnt think about that, granted its only ever 1 listener, so it should be fast, but no reason to keep the lock for that.